### PR TITLE
Multiple CDC tables from single source

### LINKED ISSUE #1482: Document: feat(cdc): share a changelog stream for multiple cdc tables
This issue tracks the documentation update needed for the merged PR #12535.

Source PR URL: https://github.com/risingwavelabs/risingwave/pull/12535
Source PR Merged At: 2023-10-31T11:19:13Z

### PR DESCRIPTION
**Pull request by emile-00**
## Info

- **Description**

  - Doc how to create multiple CDC tables from a single source and update CDC sources. 

- **Notes**

  - [ Include any supplementary context or references here. ]

- **Related code PR**

  - https://github.com/risingwavelabs/risingwave/pull/12535

- **Related doc issue**
  
  Resolves https://github.com/risingwavelabs/risingwave-docs/issues/1482

## For reviewers

- **Preview**

  - [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **dev** version of the documentation. ]

- **Key points**

  - [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`CharlieSYH`, `emile-00`, & `hengm3467`).

### CODE REVIEW

**Comment by emile-00** [comment]
@StrikeW Please review the doc updates when you get the chance, thanks!

File: docs/ingest/ingest-from-cdc.md [hunk 1]
```diff
@@ -69,6 +69,52 @@ Please see the respective data ingestion pages for the connection parameters.
...
+
+RisingWave supports creating multiple CDC tables that share a single source. 
+
+To enable this feature, the `cdc_backfill` variable to `true`.
+
+```sql
+SET cdc_backfill = "true";
+```
+
+Connect to the upstream database by creating a CDC source using the [`CREATE SOURCE`](/sql/commands/create-source.md) command. The data format is fixed as `FORMAT PLAIN ENCODE JSON` so it does not need to be specified. While a MySQL database is used as an example, any CDC source will work.
```
Comment (StrikeW) [inline docs/ingest/ingest-from-cdc.md]: We should note down here this new syntax only supports for `mysql-cdc` right now, and `postgres-cdc` is still in progress.

File: docs/ingest/ingest-from-cdc.md [hunk 2]
```diff
@@ -69,6 +69,52 @@ Please see the respective data ingestion pages for the connection parameters.
...
 - [Kinesis](ingest-from-kinesis.md)
 
+## Create multiple CDC tables with the same source
+
+RisingWave supports creating multiple CDC tables that share a single source. 
+
+To enable this feature, the `cdc_backfill` variable to `true`.
+
+```sql
+SET cdc_backfill = "true";
```
Comment (StrikeW) [inline docs/ingest/ingest-from-cdc.md]: This session config has been removed (https://github.com/risingwavelabs/risingwave/pull/13765), we should update the related pages

File: docs/guides/ingest-from-citus-cdc.md [hunk 3]
```diff
@@ -86,6 +86,10 @@ Unless specified otherwise, the fields listed are required.
 |slot.name| Optional. The slot name for each source. Each source should have a unique slot name.|
 |transactional| Optional. Specify whether you want to enable transactions for the CDC table that you are about to create. Transactions within a CDC table is an experimental feature. For details, see [Transaction within a CDC table](/concepts/transactions.md#transactions-within-a-cdc-table).|
 
+### Multiple CDC tables
+
+For more information on how to create multiple CDC tables from a single Citus source, see [Create multiple CDC tables with the same source](/ingest/ingest-from-cdc.md#create-multiple-cdc-tables-with-the-same-source)
```
Comment (StrikeW) [inline docs/guides/ingest-from-citus-cdc.md]: Citus doesn't support the new syntax to create multiple cdc tables.

File: docs/guides/ingest-from-mongodb-cdc.md [hunk 4]
```diff
@@ -85,3 +85,7 @@ WITH (
 ```
 
 After the table is created, you can view and transform the data based on your needs.
+
+### Multiple CDC tables
+
+For more information on how to create multiple CDC tables from a single MongoDB source, see [Create multiple CDC tables with the same source](/ingest/ingest-from-cdc.md#create-multiple-cdc-tables-with-the-same-source)
```
Comment (StrikeW) [inline docs/guides/ingest-from-mongodb-cdc.md]: ditto

File: docs/ingest/ingest-from-cdc.md (hunk 1 above)
Comment (StrikeW) [reply docs/ingest/ingest-from-cdc.md]: And the new functionality provide incremental and lock-free snapshot to users, that is when they are loading the historical data from upstream, the process can be recoverable if the cluster is crashed in the middle.

Comment (StrikeW) [reply docs/ingest/ingest-from-cdc.md]: And user can check the progress of backfilling historical data by querying the internal table (find the corresponding internal table by `show internal tables`). For example, here is an exmaple of quering the progress of a CDC table named `orders_rw`;
```sql
dev=> select * from __internal_orders_rw_4002_streamcdcscan_5002;
-[ RECORD 1 ]-----+---------------------------------------------------------------
split_id          | 5001
o_orderkey        | 4024320
backfill_finished | f
row_count         | 1006080
cdc_offset        | {"MySql": {"filename": "binlog.000005", "position": 60946679}}
```

**Review by StrikeW — APPROVED** [review]
LGTM

**Review by hengm3467 — APPROVED** [review]
LGTM. Thanks.

File: docs/guides/ingest-from-mysql-cdc.md [hunk 5]
```diff
@@ -197,6 +197,59 @@ All the fields listed below are required.
...
+    username = 'root',
+    password = '123456',
+    database.name = 'mydb',
+    server.id = 5888
+);
+```
+
+With the source created, you can create multiple CDC tables that ingests data from different tables in the upstream database without needing to specify the database connection parameters again. 
+
+For instance, the following CDC table in RisingWave ingests data from table `t1` in database `mydb`. When specifying the MySQL table name in the `FROM` clause after the keyword `TABLE`, the database name must also be qualified.
```
Comment (hengm3467) [inline docs/guides/ingest-from-mysql-cdc.md]:
```suggestion
For instance, the following CDC table in RisingWave ingests data from table `t1` in database `mydb`. When specifying the MySQL table name in the `FROM` clause after the keyword `TABLE`, the database name must also be specified. 
```

**Pull request merged**